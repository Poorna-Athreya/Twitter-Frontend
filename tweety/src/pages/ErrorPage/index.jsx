import React from 'react';
import PropTypes from 'prop-types';

function ErrorPage({ errorName, errorCode }) {
  return (
    <div className="not-found">
      Error
      {' '}
      {errorCode}
      !
      {' '}
      {errorName}
    </div>
  );
}
ErrorPage.propTypes = {
  errorName: PropTypes.string.isRequired,
  errorCode: PropTypes.string,
};
ErrorPage.defaultProps = {
  errorCode: '',
};
export default ErrorPage;
