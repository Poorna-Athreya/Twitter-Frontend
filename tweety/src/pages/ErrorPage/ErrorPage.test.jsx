import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorPage from './index';

describe('ErrorPage', () => {
  it('should take a snapshot of ErrorPage', () => {
    const { asFragment } = render(<ErrorPage />);

    expect(asFragment(<ErrorPage />)).toMatchSnapshot();
  });
  it('should print the correct error message when the correct prop types are given', () => {
    const mockErrorName = 'Bad Request';
    const mockErrorCode = '400';
    render(<ErrorPage errorName={mockErrorName} errorCode={mockErrorCode} />);
    expect(screen.getByTestId('errorText')).toBeTruthy();
    expect(screen.queryAllByText(`Error${mockErrorCode}!${mockErrorName}`)).toBeTruthy();
  });
});
