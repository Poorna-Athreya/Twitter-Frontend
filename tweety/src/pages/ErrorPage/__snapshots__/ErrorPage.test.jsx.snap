// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should take a snapshot of ErrorPage 1`] = `
<DocumentFragment>
  <div
    class="not-found"
    data-testid="errorText"
  >
    Error ! Something Went Wrong!
  </div>
</DocumentFragment>
`;
